Cabal/tests/UnitTests/Distribution/Version.hs

```diff
@@ -104,23 +104,22 @@ versionTests =
 --     -- , property prop_parse_disp5
 --   ]
 
-#if !MIN_VERSION_QuickCheck(2,9,0)
+adjustSize :: (Int -> Int) -> Gen a -> Gen a
+adjustSize adjust gen = sized (\n -> resize (adjust n) gen)
+
 instance Arbitrary Version where
   arbitrary = do
     branch <- smallListOf1 $
                 frequency [(3, return 0)
                           ,(3, return 1)
                           ,(2, return 2)
                           ,(1, return 3)]
-    return (Version branch []) -- deliberate []
+    return (mkVersion branch) -- deliberate []
     where
       smallListOf1 = adjustSize (\n -> min 5 (n `div` 3)) . listOf1
 
-  shrink (Version branch []) =
-    [ Version branch' [] | branch' <- shrink branch, not (null branch') ]
-  shrink (Version branch _tags) =
-    [ Version branch [] ]
-#endif
+  shrink ver = [ mkVersion branch' | branch' <- shrink (unVersion ver)
+                                   , not (null branch') ]
```

Comment:
Again, I think dropping base's Version with it's funny names of `versionBranch` and `versionTags` gives us an opportunity to use a naming convention that fits better. It's really not a branch. I think digits is the better name for what a version consists of.